### DISCUSSION TITLE
Is this possible to sample different numbers of patches from different images depending on the image size

### DISCUSSION THREAD

**Discussion by jxchen01** [Project-MONAI/MONAI#5360] [Q&A]
When using random crop sampling, currently we need to set how many samples we want to get. Is it possible to support different numbers of samples depending on the image size? Large image gets more samples, while smaller image get fewer samples?

**Comment by KumoLiu** [comment]
Hi @jxchen01, thanks for this proposal. I'm not sure if this would affect the results obviously and if there is a need for it. Maybe you have experimental proof or papers on the benefits of doing this and could share them with us. Perhaps later we could consider adding it. Thanks in advance!

**Reply by jxchen01** [reply]
this comes from one of my current projects. It is a microscopy image segmentation problem. But, in the training set, we can images of very different sizes, like 60 x 300 x 300, 70 x 500 x 500, 100 x 1500 x 1500, 120 x 2000 x 2000, etc.. Suppose we always sample patches of size 32 x 256 x 256 from each image, it is reasonable to sample maybe 2 from small images, but it would be better to sample a lot more from those larger images. Otherwise, in some sense, the training will be bias towards smaller images.

**Reply by KumoLiu** [reply]
Hi @jxchen01, If your image is the same scale, training may not actually bias toward smaller images. But in your case, you could also upsample the bigger data to enable larger data to be sampled more often. For different sizes of data, taking different numbers of samples will result in different batch sizes for each iteration, which may also have an impact on the results. 
cc @wyli @Nic-Ma
